Building with Clang on Windows Results in a Build Error

Maybe we're doing something wrong, but when building with Clang on Windows 11 using LLVM 19.1.7, we get the following output in our project that depends on QCoro:

```
[build] <redacted>/_deps/qcoro-src/qcoro/core\impl/isqprivatesignal.h:42:16: error: extension used [-Werror,-Wlanguage-extension-token]
[build]    42 |         return __FUNCSIG__;
[build]       |                ^
[build] 1 error generated.
```

Are `-Werror, -Wlanguage-extension-token` being added to our project because `QCoro` sets them? We're currently setting these flags:

```cmake
target_compile_options(
    ${PROJECT_NAME}
    PRIVATE $<$<COMPILE_LANG_AND_ID:CXX,Clang,GNU>:
            -Wall
            -Wpedantic
            -Wextra
            -Wno-pragma-once-outside-header
            -Wno-language-extension-token
            >
)
```

Note that using `-Wno-language-extension-token` has no effect.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,7 +125,7 @@ if (MSVC)
     endif()
 else()
     # Only enable strict warnings in debug mode
-    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Werror -pedantic")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wextra -Werror -pedantic -Wno-language-extension-token")
 
     if (ANDROID)
         include(DetectAndroidNDKVersion)

